chevah/compat
# [#3983] Add support for folder iterator API.

Scope
=====

This is built on top of scandir module to allow a non-blocking / iterative way of listing a directory.


Changes
=======

the paver.sh and the whole build system was udpated to use the python binary archives based on versions.

Wrapped the scandir function to a compat like method.
It convers some Windows error to Unix one for simple handling.

As a drive by I have moved all the assertions into a separate code.

As a drive by the mk.bytes helper was updated to return a native bytes() and not the specific bytesarray object.

As a drive by, I have updated `paver deps` to not install the whole build dependecies on buildslaves, as this is not needed.

BUILD_PACKAGES are needed for PQM, test_review... etc, but the buildslaves are not doing this.... so there is no need to pull a lot of dependencies..
We can break this further and only have pylint/pyflakses dependencies installed on os-independent

I have created a few test  filesystem helpers which are creating folders with auto-cleanup.


How to try and test the changes
===============================

reviewers: @alibotean @brunogola 

check that changes make sense.

sorry for the long diff, but otherwise I would never done all this cleanup

thanks!
